riscv/insns/fadd_h.h

```diff
@@ -1,5 +1,10 @@
-require_extension(EXT_ZFH);
-require_fp;
+require_either_extension(EXT_ZFH, EXT_ZHINX);
+require_fp_or_zfinx;
 softfloat_roundingMode = RM;
-WRITE_FRD(f16_add(f16(FRS1), f16(FRS2)));
+if(p->extension_enabled(EXT_ZFINX)) {
+  WRITE_FRD_H(f16_add(f16(FRS1_H), f16(FRS2_H)).v);
+} else {
+  WRITE_FRD(f16_add(f16(FRS1), f16(FRS2)));
+}
+
```

Comment:
This is really unfortunate, having to duplicate logic in each and every FP instruction. There's got to be a better way.

Couldn't the `FRS1`, `WRITE_FRD`, etc macros consider Zfinx and do the right thing?

Reply:
Maybe we can reuse FRS1/2/3_H/S/D and WRITE_FRD_H/S/D in normal case. Change the above code into :
 `WRITE_FRD_H(f16_add(FRS1_H, FRS2_H));`
 is this OK?

Reply:
OK.